whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageDequeue.java

```diff
@@ -272,9 +274,19 @@ private void handleMessageInContext(final String messageId, final MimeMessage me
 
     private void handleUpdates(final MailMessage mailMessage, final UpdateContext updateContext) {
         final List<Update> updates = updatesParser.parse(updateContext, mailMessage.getContentWithCredentials());
+        addWarnIfPasswordExists(updateContext, updates);
 
         final UpdateRequest updateRequest = new UpdateRequest(mailMessage, mailMessage.getKeyword(), updates);
         final UpdateResponse response = messageHandler.handle(updateRequest, updateContext);
         mailGateway.sendEmail(mailMessage.getReplyToEmail(), response.getStatus() + ": " + mailMessage.getSubject(), response.getResponse(), null);
     }
+
+    private static void addWarnIfPasswordExists(final UpdateContext updateContext, final List<Update> updates) {
+        for (Update update : updates) {
+            if (!update.getCredentials().ofType(PasswordCredential.class).isEmpty()){
+                updateContext.addGlobalMessage(UpdateMessages.passwordInMailUpdate());
```

Comment:
will it add multiple global messages or single one incase more than one update happened

Reply:
single one. It will return; once the global message is added

Reply:
I will create a test for this

Reply:
I checked it using a set so should be good